pkg/kubelet/secret/secret_manager.go

```diff
@@ -201,8 +201,12 @@ type cachingSecretManager struct {
 }
 
 func NewCachingSecretManager(kubeClient clientset.Interface) (Manager, error) {
+	// TODO: Currently apiserver completely cannot keep up with 1 minute ttl in
+	// 5000-node clusters. So before we optimize apiserver and/or etcd, increase
+	// the TTL to 5 minutes (which is a lot, but...)
+	ttl := 5 * time.Minute
```

Comment:
> within a single node, almost every pod comes from a different namespace

Hm...that's interesting. In that case, I don't think that adding secret manager (with the default 1min TTL) would've helped at all. By default, kubelet syncs every pod every minute. If almost all of them are from different namespaces, the caching secret manager should make almost no difference...